tracer/test/Datadog.Trace.IntegrationTests/LibDatadog/TraceExporterTests.cs

```diff
@@ -0,0 +1,162 @@
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.AppSec.Rasp;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.IntegrationTests.LibDatadog;
+
+public class TraceExporterTests
+{
+    [SkippableTheory]
+    [InlineData(TestTransports.Tcp)]
+    [InlineData(TestTransports.Uds)]
+    [InlineData(TestTransports.WindowsNamedPipe)]
+    public async Task SendsTracesUsingDataPipeline(TestTransports transport)
+    {
+        if (EnvironmentTools.IsWindows() && Environment.Is64BitProcess == false)
+        {
+            throw new SkipException("Can't use data pipeline on win-x86, there is no good way to copy native libraries for win-x86");
+        }
+
+        if (transport == TestTransports.WindowsNamedPipe && !EnvironmentTools.IsWindows())
+        {
+            throw new SkipException("WindowsNamedPipe transport is only supported on Windows");
+        }
+
+        if (transport == TestTransports.Uds && !EnvironmentTools.IsLinux())
+        {
+                throw new SkipException("Unix Domain Sockets (UDS) transport is only supported on Linux when data pipeline is enabled");
+        }
+
+        var pipeName = $"trace-{Guid.NewGuid()}";
+        var udsPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        using var agent = GetAgent();
+        var settings = GetSettings();
+        var tracerSettings = TracerSettings.Create(settings);
+
+        agent.CustomResponses[MockTracerResponseType.Traces] = new MockTracerResponse
+        {
+            StatusCode = 200,
+            ContentType = "application/json",
+            Response = """
+                       {
+                           "rate_by_service": {
+                               "service:default-service,env:test": 1.0,
+                               "service:,env:": 0.8
+                           }
+                       }
+                       """
+        };
+
+        var discovery = DiscoveryService.Create(tracerSettings.Exporter);
+        var tracer = TracerHelper.Create(tracerSettings, discoveryService: discovery);
+
+        using var span = tracer.StartSpan("operationName");
+        span.ResourceName = "resourceName";
+        span.Type = "test";
+
+        var testMetaStruct = new TestMetaStruct
+        {
+            Foo = "foo",
+            Bar = 1,
+        };
+
+        var metaStructBytes = MetaStructHelper.ObjectToByteArray(testMetaStruct.ToDictionary());
+        span.SetMetaStruct("test-meta-struct", metaStructBytes);
+        span.Finish();
+
+        await tracer.TracerManager.ShutdownAsync();
+        var recordedSpans = agent.WaitForSpans(1);
+        recordedSpans.Should().ContainSingle();
+
+        var recordedSpan = recordedSpans[0];
```

Comment:
nit: you can combine these lines
```suggestion
        var recordedSpan = recordedSpans.Should().ContainSingle().Subject;
```